packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -167,3 +170,17 @@ function buildGlobPattern(ast: any) {
   }
   return pattern
 }
+
+function getQueryDelimiterIndex(rawUrl: string): number {
+  let bracketsStack = 0
+  for (let i = 0; i < rawUrl.length; i++) {
+    if (rawUrl[i] === '{') {
+      bracketsStack++
+    } else if (rawUrl[i] === '}') {
+      bracketsStack--
+    } else if (rawUrl[i] === '?' && bracketsStack === 0) {
+      return i
+    }
+  }
+  return -1
```

Comment:
I think this may be better than a regex here. I was thinking we could check for `${` but this may be enough.